Services/DataBaseService/Repositories/UserCredentialRepository.cs

```diff
@@ -0,0 +1,36 @@
+﻿using DataBaseService.DbModels;
+using DataBaseService.Interfaces;
+using DTO;
+using Microsoft.Data.SqlClient;
+using System;
+
+namespace DataBaseService.Repositories
+{
+    public class UserCredentialRepository : IRepository<UserEmailPassword>
+    {
+        private readonly IMapper<UserEmailPassword, DbUserCredential> mapper;
+        private readonly DataBaseContext dbContext;        
+        
+        public UserCredentialRepository(IMapper<UserEmailPassword, DbUserCredential> mapper, DataBaseContext dbContext)
+        {
+            this.mapper = mapper;
+            this.dbContext = dbContext;
+        }
+
+        public void Create(UserEmailPassword data)
+        {
+            //TODO change to custom exception
+            try
+            {
+                dbContext.Add(mapper.CreateMap(data));
+                dbContext.SaveChanges();
+            }
+            catch(SqlException e)
```

Comment:
Is there a possibility that here might occur some other type od exception (I personally don't know, so just asking)

Reply:
As I now, there can only be such exceptions.